cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,364 @@
+});
+
+// POST endpoint `/products` to create a new product.
+app.post("/products", (req, res) => {
+  const newProduct = req.body.product_name;
+
+  if (!newProduct) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "Product not specified",
+    });
+  }
+
+  pool
+    .query("SELECT * FROM products WHERE product_name=$1", [newProduct])
+    .then((result) => {
+      if (result.rows.length > 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `The product ${newProduct} already exists`,
+        });
+      } else {
+        const query = "INSERT INTO products (product_name) VALUES($1)";
+        pool
+          .query(query, [newProduct])
+          .then(() => res.json({ msg: "The new product was added successful" }))
+          .catch((error) => {
+            res.status(400).json(error);
+          });
+      }
+    })
+    .catch((error) => res.status(400).json(error));
+});
+
+//Add a new POST endpoint `/availability` to create a new product availability (with a price and a supplier id). Check that the price is a positive integer and that both the product and supplier ID's exist in the database, otherwise return an error
+
+app.post("/availability", (req, res) => {
+  const price = req.body.unit_price;
+  const productId = req.body.prod_id;
+  const supplierId = req.body.supp_id;
+
+  if (!Number.isInteger(price) && price <= 0) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "Invalid input. The price should be a positive number",
+    });
+  }
+
+  pool
+    .query(
+      `SELECT product_availability.prod_id, product_availability.supp_id 
+      FROM product_availability 
+      INNER JOIN products ON products.id = product_availability.prod_id 
+      INNER JOIN suppliers ON suppliers.id = product_availability.supp_id 
+      WHERE product_availability.prod_id = $1 AND product_availability.supp_id = $2`,
+      [productId, supplierId]
+    )
+    .then((result) => {
+      if (result.rows.length == 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `Id not found`,
+        });
+      } else {
+        const query = `SELECT prod_id, supp_id FROM product_availability WHERE prod_id = $1 AND supp_id = $2`;
+        pool.query(query, [productId, supplierId]).then((result) => {
+          if (result.rows.length > 0) {
+            return res.status(400).json({
+              result: "failure",
+              msg: "This product availability already exists",
+            });
+          } else {
+            const query =
+              "INSERT INTO product_availability (prod_id, supp_id, unit_price) VALUES($1, $2, $3)";
+            pool
+              .query(query, [productId, supplierId, price])
+              .then(() =>
+                res.json({
+                  msg: "A new product availability was added successful",
+                })
+              )
+              .catch((error) => {
```

Comment:
The catch you placed at the end is sufficient for the whole chain, I believe, will have to double check on this, otherwise I have so much to change in my code lol